nautobot/dcim/tests/features/locations.feature

```diff
@@ -0,0 +1,143 @@
+
+  Background:
+    Given I have appropriate Nautobot permissions:
+      | model             | permissions            |
+      | dcim.locationtype | add,change,delete,view |
+      | dcim.location     | add,change,delete,view |
+    And I am logged in to Nautobot
+
+  Scenario: Creating a hierarchy of LocationTypes
+    When I navigate to the "Add LocationType" view
+    And I make the following form entries:
+      | name              | nestable |
+      | Geographic Region | True     |
+    And I submit the form
+    Then a LocationType exists in Nautobot with the following properties:
+      | name              | parent | nestable |
+      | Geographic Region | None   | True     |
+    And Nautobot shows the "LocationType Detail" view for the LocationType with name "Geographic Region"
+
+    When I click the "Add child" button
+    Then Nautobot shows the "Add LocationType" view with the following presets:
+      | parent            |
+      | Geographic Region |
+    When I make the following form entries:
+      | name |
+      | city |
+    And I submit the form
+    Then a LocationType exists in Nautobot with the following properties:
+      | name | parent              | nestable |
+      | City | Geographic Region   | False    |
+    And Nautobot shows the "LocationType Detail" view for the LocationType with name "City"
+
+    When I click the "Add child" button
+    Then Nautobot shows the "Add LocationType" view with the following presets:
+      | parent |
+      | City   |
+    When I make the following form entries:
+      | name     | content_types |
+      | Building | [dcim.device] |
+    And I submit the form
+    Then a LocationType exists in Nautobot with the following properties:
+      | name     | parent | nestable | content_types |
+      | Building | City   | False    | [dcim.device] |
+    And Nautobot shows the "LocationType Detail" view for the LocationType with name "Building"
+
+  Scenario: Creating a hierarchy of Locations
+    Given that LocationType records exist with the following properties:
+      | name              | parent            | nestable |
+      | Geographic Region | None              | True     |
+      | City              | Geographic Region | False    |
+      | Building          | Building          | False    |
+    When I navigate to the "Add Location" view
+    And I make the following form entries:
+      | name          | location_type     | status |
+      | United States | Geographic Region | Active |
+    And I submit the form
+    Then a Location exists in Nautobot with the following properties:
+      | name          | location_type     | parent | status |
+      | United States | Geographic Region | None   | Active |
+    And Nautobot shows the "Location Detail" view for the Location with name "United States"
+
+    When I click the "Add child" button
+    Then Nautobot shows the "Add Location" view with the following presets:
+      | parent        |
+      | United States |
```

Comment:
~Do we need to add a `location_type` preset here as well or no?~
NVM you still have to pick from the dropdown